trl/mergekit_utils.py

```diff
@@ -0,0 +1,347 @@
+import os
+import torch
+
+from mergekit.config import MergeConfiguration
+from mergekit.merge import MergeOptions, run_merge
+
+
+from huggingface_hub import HfApi
+
+
+def get_last_checkpoint_path(output_dir):
+    """
+    Get the path to the most recent checkpoint in the output_dir.
+    
+    Args:
+        output_dir (str): The directory where the checkpoints are saved.
+    
+    Returns:
+        str: The path to the most recent checkpoint, or None if no checkpoint exists.
+    """
+    # Ensure the output directory exists
+    if not os.path.exists(output_dir):
+        print(f"Output directory '{output_dir}' does not exist.")
+        return None
+    
+    # Get all subdirectories in the output directory
+    checkpoints = [
+        os.path.join(output_dir, d) for d in os.listdir(output_dir)
+        if os.path.isdir(os.path.join(output_dir, d)) and "checkpoint" in d
+    ]
+    
+    # If no checkpoints are found
+    if not checkpoints:
+        print("No checkpoints found.")
+        return None
+    
+    # Sort checkpoints by their last modification time
+    last_checkpoint = max(checkpoints, key=os.path.getmtime)
+    
+    return last_checkpoint
+
+
+def upload_model_to_hf(folder_path, repo_id):
+    api = HfApi()
+    # Create the repository if it doesn't exist
+    repo = api.create_repo(repo_id, repo_type="model")
+    
+    # Upload the folder to the specified repository
+    api.upload_folder(
+        folder_path=folder_path,
+        repo_id=repo.repo_id,
+        repo_type=repo.repo_type,
+    )
+
+class MergeConfig:
```

Comment:
Could you please add a docstring like we do for our model training configs: https://github.com/huggingface/trl/blob/623963126be5598bd1eea4ec82b43447fcc11535/trl/trainer/dpo_config.py#L34

This way it renders nicely in the docs :)

It would be good to provide some links to the various supported methods

Reply:
Useful ressource: https://github.com/huggingface/trl/pull/1944#issue-2471984781

Reply:
I got GPT to add the docstring. 
is it ok?

Reply:
Sure!